.cirrus.yml

```
@@ -0,0 +1,100 @@
+pub_cache: &pub_cache
+  pub_cache:
+    folder: ~/.pub-cache
+    fingerprint_script: cat pubspec.lock
+
+gen_gradle_cache_key: &gen_gradle_cache_key
+  gen_gradle_cache_key_script: 
+    - bash ci/gradle_checksum.sh gradle_checksum.txt
+    - ls
+
+gradle_cache: &gradle_cache
+  gradle_cache:
+    folder: ~/.gradle/caches
+    fingerprint_script: 
+      - ls
+      - echo gradle-{{ checksum "gradle_checksum.txt" }}-{{ checksum "android/gradle/wrapper/gradle-wrapper.properties" }}
```

Cirrus doesn't support Circle CI style caching with `checksum`. You can simply run something like this:

```suggestion
      - cat gradle_checksum.txt android/gradle/wrapper/gradle-wrapper.properties
```

Please also refer to documentation around [caching Gradle](https://cirrus-ci.org/examples/#caching).

BTW you might want to skip the `fingerprint_script` part since Gradle cache content is pretty undeterministic. 😪

---

Thank you @fkorotkov, I had separated the cache for the `.gradle/caches` and `.gradle/wrapper`, and I think don't need the checksum now.